File: Formula-Bollo/src/shared/models/fantasyPointsDriver.ts
import { Driver } from "./driver";
import { Race } from "./race";
import { Season } from "./season";

export class FantasyPointsDriver {
  id: number;
  driver: Driver;
  race: Race;
  points: number;
  season: Season;
  totalPoints?: number;

  constructor(id: number, driver: Driver, race: Race, points: number, season: Season) {
    this.id = id;
    this.driver = driver;
    this.race = race;
    this.points = points;
    this.season = season;
  }
}

File: Formula-Bollo-App/src/main/java/formula/bollo/app/entity/FantasyPointsDriver.java
package formula.bollo.app.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;


@Entity
@Table(name = "fantasy_points_driver")
@Getter
@Setter
@ToString
@EqualsAndHashCode
@AllArgsConstructor
@NoArgsConstructor
public class FantasyPointsDriver {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @ManyToOne
    @JoinColumn(name = "driver_id", referencedColumnName = "id")
    private Driver driver;

    @ManyToOne
    @JoinColumn(name = "race_id", referencedColumnName = "id")
    private Race race;

    @Column(name = "points", nullable = false, length = 3)
    private int points;

    @ManyToOne
    @JoinColumn(name = "season_id", referencedColumnName = "id")
    private Season season;
}
